pkg/auth/oauth/dynamic_registration.go

```diff
@@ -27,17 +27,40 @@ const ResponseTypeCode = "code"
 // TokenEndpointAuthMethodNone is the token endpoint auth method for none
 const TokenEndpointAuthMethodNone = "none"
 
+// RequestScopeList represents the "scope" field in a dynamic client registration request.
+// Per RFC 7591 Section 2, this must be marshaled as a space-delimited string, not a JSON array.
+//
+// Examples of marshaled output:
+//
+//	[]string{"openid", "profile", "email"}  → "openid profile email"
+//	[]string{"openid"}                      → "openid"
+//	nil or []string{}                       → omitted (via omitempty)
+type RequestScopeList []string
+
+// MarshalJSON implements custom encoding for RequestScopeList. It converts the slice
+// of scopes into a space-delimited string as required by RFC 7591 Section 2.
+func (r RequestScopeList) MarshalJSON() ([]byte, error) {
+	// Handle nil or empty slice - return null so omitempty works
+	if len(r) == 0 {
+		return []byte("null"), nil
```

Comment:
is this to make Go's omitempty work?

Reply:
Excellent question.  Ive subsequently dug into omitempty's implementation a bit further to understand how it would interact with this code, and was able to be simplify.